### ISSUE TITLE
IP-1593. Big Data > Implement BigQuery to store data

### ISSUE THREAD

**Issue by idea2-pepito** [idea2-platform/pepito-test#970]
Next step is to allow to configure Google BigQuery as a big data service in App Settings and sync data to this service. So it means that entities and fields that have big data enabled should be synced to BigQuery. This includes the following things:
When new data is added to an entity where big data is enabled (or in one of the fields inside the entity), data should be appended to a BigQuery table.
If data is modified, we should append a new record to BigQuery with a different timestamp.
If data is deleted, we should append a new record to BigQuery with a different timestamp and a flag indicating the record was deleted.
If we change data structure in Idea2, we should perform the refactor in BigQuery. Keep in mind that this could be very specific to each service, so keep the interface quite generic.
Be able to configure the frequency we will sync data to BigQuery. Could be immediate, but maybe we want to do it only once a day.

So here we will be making sure that we put the information we need in BigQuery, but we won't do anything with it yet. That's going to be done in the following step.

Finally, it is very important to keep in mind that most of these big data analysis services do not allow to change/delete records in the dataset. It means we can only append data so we should support multiple versions using a timestamp field and a delete flag. Then queries should take this into account to use only the latest versions of the records.
